test/cpp/util/channelz_sampler.cc

```diff
@@ -0,0 +1,289 @@
+    std::queue<grpc::channelz::v1::Subchannel>& subchannel_queue) {
+  GetSubchannelRequest get_subchannel_request;
+  get_subchannel_request.set_subchannel_id(subchannel_id);
+  GetSubchannelResponse get_subchannel_response;
+  ClientContext get_subchannel_context;
+  channelz_stub->GetSubchannel(&get_subchannel_context, get_subchannel_request,
+                               &get_subchannel_response);
+  subchannel_queue.push(get_subchannel_response.subchannel());
+}
+
+void GetChannelDescedence(
+    grpc::channelz::v1::Channelz::Stub* channelz_stub,
+    grpc::channelz::v1::Channel& channel,
+    std::queue<grpc::channelz::v1::Channel>& channel_queue,
+    std::queue<grpc::channelz::v1::Subchannel>& subchannel_queue) {
+  std::cout << "    Channel " << GetChannelID(channel) << " descendence - ";
+  if (channel.channel_ref_size() > 0) {
+    std::cout << "channel: ";
+    for (auto& i : channel.channel_ref()) {
+      int64_t ch_id = i.channel_id();
+      std::cout << ch_id << " ";
+      GetChannelRPC(ch_id, channelz_stub, channel_queue);
+    }
+  }
+  if (channel.subchannel_ref_size() > 0) {
+    std::cout << "subchannel: ";
+    for (auto& i : channel.subchannel_ref()) {
+      int64_t subch_id = i.subchannel_id();
+      std::cout << subch_id << " ";
+      GetSubchannelRPC(subch_id, channelz_stub, subchannel_queue);
+    }
+  }
+  if (channel.socket_ref_size() > 0) {
+    std::cout << "socket: ";
+    for (auto& i : channel.socket_ref()) {
+      int64_t so_id = i.socket_id();
+      std::cout << so_id << " ";
+    }
+  }
+  std::cout << std::endl;
+}
+
+void GetSubchannelDescedence(
+    grpc::channelz::v1::Channelz::Stub* channelz_stub,
+    grpc::channelz::v1::Subchannel& subchannel,
+    std::queue<grpc::channelz::v1::Channel>& channel_queue,
+    std::queue<grpc::channelz::v1::Subchannel>& subchannel_queue) {
+  std::cout << "    Subchannel " << GetSubchannelID(subchannel)
+            << " descendence - ";
+  if (subchannel.channel_ref_size() > 0) {
+    std::cout << "channel: ";
+    for (auto& i : subchannel.channel_ref()) {
+      int64_t ch_id = i.channel_id();
+      std::cout << ch_id << " ";
+      GetChannelRPC(ch_id, channelz_stub, channel_queue);
+    }
+  }
+  if (subchannel.subchannel_ref_size() > 0) {
+    std::cout << "subchannel: ";
+    for (auto& i : subchannel.subchannel_ref()) {
+      int64_t subch_id = i.subchannel_id();
+      std::cout << subch_id << " ";
+      GetSubchannelRPC(subch_id, channelz_stub, subchannel_queue);
+    }
+  }
+  if (subchannel.socket_ref_size() > 0) {
+    std::cout << "socket: ";
+    for (auto& i : subchannel.socket_ref()) {
+      int64_t so_id = i.socket_id();
+      std::cout << so_id << " ";
+    }
+  }
+  std::cout << std::endl;
+}
+
+int main(int argc, char** argv) {
+  // make sure flags can be used
+  grpc::testing::InitTest(&argc, &argv, true);
+  std::cout << "server address: " << FLAGS_server_address << std::endl;
```

Comment:
nit: lets remove printing of flags back to stdout -- I think it will likely clutter the output without too much benefit here

Reply:
done